libs/saksbehandling-common/src/main/kotlin/person/Folkeregisteridentifikator.kt

```diff
@@ -17,19 +17,21 @@ import java.time.temporal.ChronoUnit
 class Folkeregisteridentifikator private constructor(
     @JsonValue val value: String,
 ) {
-    init {
-        require(FolkeregisteridentifikatorValidator.isValid(value))
-    }
-
     companion object {
         @JvmStatic
         @JsonCreator
-        fun of(fnr: String?): Folkeregisteridentifikator =
-            try {
-                Folkeregisteridentifikator(fnr!!.replace(Regex("[^0-9]"), ""))
-            } catch (e: Exception) {
-                throw InvalidFoedselsnummerException(fnr, e)
+        fun of(fnr: String?): Folkeregisteridentifikator {
+            if (fnr.isNullOrEmpty()) {
+                throw InvalidFoedselsnummerException("Fnr er tomt")
+            } else {
+                val fnrMedGyldigeTall = fnr.replace(Regex("[^0-9]"), "")
+                if (FolkeregisteridentifikatorValidator.isValid(fnrMedGyldigeTall)) {
+                    return Folkeregisteridentifikator(fnrMedGyldigeTall)
+                } else {
+                    throw InvalidFoedselsnummerException("Fødselsnummeret er ugyldig")
```

Comment:
Exception-klassa forventar fnr som første string-param, og her sender du det som vel meir er exception-message, det skurrar litt.

Reply:
Utover det ei fin endring 👍

Reply:
Jepp, ble litt rart det der. Endret så vi sender med detaljer beskrivelsesfeltet i stedet for og logger det i `loggForespoerselException`.